modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/SecondaryStoreUserLoginTestCase.java

```diff
@@ -0,0 +1,269 @@
+    private static final UserStoreConfigUtils userStoreConfigUtils = new UserStoreConfigUtils();
+    private UserManagementClient userMgtClient;
+    private static final String PERMISSION_LOGIN = "/permission/admin/login";
+    private static final String JDBC_CLASS = "org.wso2.carbon.user.core.jdbc.UniqueIDJDBCUserStoreManager";
+    private static final String DOMAIN_ID = "WSO2TEST.COM";
+    private static final String USER_STORE_DB_NAME = "SECONDARY_USER_STORE_DB";
+    private Tomcat tomcat;
+    public static final int TOMCAT_PORT = 8490;
+    public static final String playgroundAppName = "playground.app";
+    public static final String playgroundAppContext = "/playground.app";
+    public static final String playgroundAppCallBackUri = "http://localhost:" + TOMCAT_PORT + "/playground" + "" +
+            ".app/oauth2client";
+    public static final String targetApplicationUrl = "http://localhost:" + TOMCAT_PORT + "%s";
+    public static final String appUserAuthorizePath = "/playground2/oauth2-authorize-user.jsp";
+    private static final Log log = LogFactory.getLog(OIDCAbstractIntegrationTest.class);
+    private static final Log LOG = LogFactory.getLog(TomcatInitializerTestCase.class);
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init();
+        CookieStore cookieStore = new BasicCookieStore();
+
+//      Register a secondary user store
+        userStoreConfigAdminServiceClient =
+                new UserStoreConfigAdminServiceClient(backendURL, sessionCookie);
+        userMgtClient = new UserManagementClient(backendURL, getSessionCookie());
+        try {
+            UserStoreDTO userStoreDTO = userStoreConfigAdminServiceClient.createUserStoreDTO(JDBC_CLASS, DOMAIN_ID,
+                    userStoreConfigUtils.getJDBCUserStoreProperties(USER_STORE_DB_NAME));
+            userStoreConfigAdminServiceClient.addUserStore(userStoreDTO);
+            Thread.sleep(5000);
+        } catch (Exception e) {
+            log.error(e);
+        }
+        boolean isSecondaryUserStoreDeployed = userStoreConfigUtils.waitForUserStoreDeployment(
+                userStoreConfigAdminServiceClient, DOMAIN_ID);
+
+        if (isSecondaryUserStoreDeployed) {
+
+//          Creating users in the primary and secondary user stores
+            try {
+                addUserIntoJDBCUserStore(primUsername, primPassword, false);
+                addUserIntoJDBCUserStore(secUsername, secPassword, true);
+            } catch (Exception e) {
+                log.error(e);
+            }
+
+//          Creating and starting application on tomcat
+            initAndCreatePlaygroundApplication();
+            startTomcat();
+            client = HttpClientBuilder.create().setDefaultCookieStore(cookieStore).build();
+
+//          To generate sessionDataKey
+            try {
+                testSendAuthenticationRequest(playgroundApp, client);
```

Comment:
why are you doing this in the before class? This should be part of the actual test.